kopf/reactor/running.py

```diff
@@ -265,9 +278,21 @@ def run(
         logger.warning(f"Root task {name!r} is finished unexpectedly.")
 
 
-async def _stop_flag_checker(should_stop):
+async def _stop_flag_checker(
+        signal_flag: asyncio.Future,
+        ready_flag: Optional[Flag],
+        stop_flag: Optional[Flag],
+):
+    # TODO: collect the readiness of all root tasks instead, and set this one only when fully ready.
+    # Notify the caller that we are ready to be executed.
+    await _raise_flag(ready_flag)
+
+    # Wait until one of the stoppers is set/raised.
     try:
-        result = await should_stop
+        aws = [signal_flag] + ([] if stop_flag is None else [_wait_flag(stop_flag)])
```

Comment:
aws?

Reply:
Awaitables, as in the docs: https://docs.python.org/3/library/asyncio-task.html#asyncio.wait

Reply:
"awaitables" looks better than "aws" IMO :P

Reply:
@samurang87 Renamed it to `flags`.